### PR TITLE
Port to core

### PR DESCRIPTION
**Pull request by Jaxx-22**
Port to NetStandard v1.6
This is backward compatible with .NetFramework v4.5 and up
Will work with .NetCore apps

### CODE REVIEW

**Comment by trailmax** [comment]
Thank you for your effort. But you have renamed the folder for the project from `Cake.SqlServer` to `SqlServer` and it is now impossible to [see what has been changed in diffs](https://github.com/AMVSoftware/Cake.SqlServer/pull/19/files#diff-2250397314ff378e2cc9dbc483c22d29L1). Please rename the folder back to what it was.

And build if run `.\build.ps1 -Target Build` is broken for me locally:

>SqlServer.xproj(8,3): error MSB4019: The imported project "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v10.0\DotNet\Microsoft.DotNet.Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.` 

Also am I right in saying that [bacpac functionality](https://github.com/AMVSoftware/Cake.SqlServer/pull/19/files#diff-0e9c4879c6785ef418307d627c777cc1R1) is not going to work on .NetStandard? Any reason for that?

**Comment by Jaxx-22** [comment]
Sorry for the rename, slight overview on my end.

And for the bacpac functionality, the code is using the dependency "Microsoft.SqlServer.DacFx.x64" : "130.3485.1" does not have a net standard build to use. So in code we look at which framework we are using.

There is still more work to be done with getting built too. I will be coming back to later on this week.

File: src/Cake.SqlServer/Cake.SqlServer.nuspec [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-    <tags>Cake Script Build SQLServer </tags>
-  </metadata>
-  <files>
-    <file src="Cake.SqlServer.dll" target="lib/net45" />
-    <file src="Cake.SqlServer.pdb" target="lib/net45" />
-    <file src="Cake.SqlServer.xml" target="lib/net45"/>
-    <file src="Microsoft.*.dll" target="lib/net45"/>
-    <file src="LICENSE" />
-  </files>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (trailmax) [inline src/Cake.SqlServer/Cake.SqlServer.nuspec]: Line endings changed? white-space? files look identical, yet diff shows 100% change

File: src/Cake.SqlServer/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-using System.Reflection;
-using System.Runtime.CompilerServices;
-
-[assembly: AssemblyProduct("Cake.SqlServer")]
-[assembly: AssemblyVersion("1.5.1.0")]
-[assembly: AssemblyFileVersion("1.5.1.0")]
-[assembly: AssemblyInformationalVersion("1.5.1-backpac.1+13.Branch.backpac.Sha.ea1d7b9246a302adb0f73eb9f55a8375480a503c")]
-[assembly: AssemblyCopyright("Copyright (c) 2016 - 2016 AMV Software")]
-
+//------------------------------------------------------------------------------
```
Comment (trailmax) [inline src/Cake.SqlServer/Properties/AssemblyInfo.cs]: Same as above. White-space?

File: src/Cake.SqlServer/SqlServerAliases.cs [hunk 3]
```diff
@@ -414,6 +414,7 @@ public static void RestoreSqlBackup(this ICakeContext context, String connection
         ///     });        
         /// </code>
         /// </example>
+#if NET451
```
Comment (trailmax) [inline src/Cake.SqlServer/SqlServerAliases.cs]: `#if NET451` should be above the XML comment, otherwise VS goes crazy

File: src/Cake.SqlServer/SqlServerAliases.cs [hunk 4]
```diff
@@ -451,7 +452,8 @@ public static void CreateBacpacFile(this ICakeContext context, String connection
         ///     	});
         ///     });        
         /// </code>
-        /// </example>        
+        /// </example>  
+#if NET451
```
Comment (trailmax) [inline src/Cake.SqlServer/SqlServerAliases.cs]: Same as above - `#if` should include xml-comment

File: src/Cake.SqlServer/SqlServerAliasesImpl.cs [hunk 5]
```diff
@@ -156,6 +156,7 @@ internal static SqlConnection OpenSqlConnection(ICakeContext context, String con
             }
             catch (SqlException exception)
             {
+#if Net541
```
Comment (trailmax) [inline src/Cake.SqlServer/SqlServerAliasesImpl.cs]: Should it not be `NET451`?

File: src/Cake.SqlServer/SqlServerAliasesImpl.cs [hunk 6]
```diff
@@ -164,6 +165,16 @@ internal static SqlConnection OpenSqlConnection(ICakeContext context, String con
                     var newException = new Exception(errorMessage, exception);
                     throw newException;
                 }
+#else
```
Comment (trailmax) [inline src/Cake.SqlServer/SqlServerAliasesImpl.cs]: I don't see a difference with the block of code above. Is `#if NET451` required here?

File: src/Tests/Program.cs [hunk 7]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Tests
```
Comment (trailmax) [inline src/Tests/Program.cs]: Is this file required?

File: src/Tests/RestoreDatabaseTests.cs [hunk 8]
```diff
@@ -172,7 +172,7 @@ public void GetDatabaseName_Should_ReturnName()
 
         private static string GetBackupFilePath(String filename = "multiFileBackup.bak")
         {
-            return Directory.GetFiles(AppDomain.CurrentDomain.BaseDirectory, filename, SearchOption.AllDirectories).FirstOrDefault();
+            return Directory.GetFiles(Directory.GetCurrentDirectory(), filename, SearchOption.AllDirectories).FirstOrDefault();
```
Comment (trailmax) [inline src/Tests/RestoreDatabaseTests.cs]: :+1:

**Comment by trailmax** [comment]
I see not much code changes are actually required to get to .NetStandard. The biggest problem is with DacFx dependency, but it caused trouble before - see #18. I'll see what can be done about this. Ability to work with `bacpac` is a pretty big thing for me and my team.

As for exception I get on build - this is due to lack of tooling installed - I need to update to the latest .Core tools, and even to VS2017. 

Exception on the build server is for packing nuget - folder structure generated by the new msbuild is different to what it was before, so `build,cake` needs to be adjusted.

**Comment by trailmax** [comment]
I'm about to go on holidays. So please excuse the lack of responsiveness for a week-or-two.

**Comment by trailmax** [comment]
Closing as lack of response
